ubuntu/playbooks/zookeepers/zookeepers.yml

```diff
@@ -424,12 +424,7 @@
         regexp: '        <property name="Connect String">.*</property>'
         line: '        <property name="Connect String">zk1.{{ name_prefix }}{{ name_suffix }}.internal:2173,zk2.{{ name_prefix }}{{ name_suffix }}.internal:2173,zk3.{{ name_prefix }}{{ name_suffix }}.internal:2173</property>'
 
-    - name: check ca exists
-      stat:
-        path: /opt/nifi-certificates/nifi-cert.pem
-      register: ca_cert
-
-    - name: get pre-existing ca from s3
+        - name: get pre-existing ca from s3
```

Comment:
I think this handles your desire cleaner.  I came across this, as Im attempting to pre-populate the cert and not have zk1 build its own


- Regardless of node, try to pull the CA
- if not there, and zk1, gen and push
- 

```
    - name: get pre-existing ca from s3
      aws_s3:
        region: "{{ aws_region }}"
        mode: get
        bucket: "{{ s3_bucket }}"
        object: "nifi/certificates/{{ item }}"
        dest: "/opt/nifi-certificates/{{ item }}"
      become_user: nifi
      ignore_errors: True
      with_items:
        - nifi-cert.pem
        - nifi-key.key

    - name: check ca exists
      stat:
        path: /opt/nifi-certificates/nifi-cert.pem
      register: ca_cert

    - name: generate ca when not exists
      shell: |
        export JAVA_HOME=/usr/lib/jvm/default-java
        /opt/nifi-toolkit/bin/tls-toolkit.sh standalone -c zk1.{{ name_prefix }}{{ name_suffix }}.internal -d 3650 -k 2048 -s SHA256WITHRSA -T jks --subjectAlternativeNames zk1.{{ name_prefix }}{{ name_suffix }}.internal,zk2.{{ name_prefix }}{{ name_suffix }}.internal,zk3.{{ name_prefix }}{{ name_suffix }}.internal,{{ lb_dns }}
      args:
        chdir: /opt/nifi-certificates/
        executable: /bin/bash
      become_user: nifi
      when:
        - ca_cert.stat.exists == false
        - ansible_nodename == 'zk1.' + name_prefix + name_suffix + '.internal'

    - name: upload ca to s3
      aws_s3:
        region: "{{ aws_region }}"
        mode: put
        bucket: "{{ s3_bucket }}"
        object: "nifi/certificates/{{ item }}"
        src: "/opt/nifi-certificates/{{ item }}"
        encrypt: yes
        encryption_mode: "aws:kms"
        encryption_kms_key_id: "{{ kms_key_id }}"
      with_items:
        - nifi-cert.pem
        - nifi-key.key
      when:
        - ansible_nodename == 'zk1.' + name_prefix + name_suffix + '.internal'
        - ca_cert.stat.exists == false
 ```